### PR TITLE
Changelog for v2.165.0

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,291 @@
...
+- Re-normalize ``risk:mitigation:name``, ``it:mitre:attack:technique:name``,
+ and ``it:mitre:attack:mitigation:name`` secondary properties.
+  (`#3585 <https://github.com/vertexproject/synapse/pull/3585>`_)
+- Re-normalize ``velocity`` properties which are float values.
+  (`#3616 <https://github.com/vertexproject/synapse/pull/3616>`_)
+- See :ref:`datamigration` for more information about automatic migrations.
+
+Model Changes
+-------------
+- Add a new model, ``sci``, for modeling scientific experiments. Updates to
```
Comment (invisig0th) [inline CHANGELOG.rst:23]:
```suggestion
- Add a new model, ``sci``, for modeling elements of the scientific method. Updates to
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,291 @@
...
+
+  ``econ:bank:swift:bic``
+    A Society for Worldwide Interbank Financial Telecommunication (SWIFT)
+    Business Identifier Code (BIC).
+
+  ``risk:vulnerable``
+    Indicates that a node is susceptible to a vulnerability.
+
+  ``sci:hypothesis:type:taxonomy``
+    FIXME MISSING DOCSTRING
```
Comment (invisig0th) [inline CHANGELOG.rst:67]: Need to smoke in some doc strings for these two FIXMEs

File: CHANGELOG.rst [hunk 3]
```diff
@@ -6,6 +6,291 @@
 Synapse Changelog
 *****************
 
+v2.165.0 - TBD
+=====================
+
+Automatic Migrations
+--------------------
+- Re-normalize ``risk:mitigation:name``, ``it:mitre:attack:technique:name``,
+ and ``it:mitre:attack:mitigation:name`` secondary properties.
```
Comment (Cisphyx) [inline CHANGELOG.rst:15]:
```suggestion
  and ``it:mitre:attack:mitigation:name`` secondary properties.
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,291 @@
...
+-------------------------
+- Change the compression mode used when streaming Cell backups to speed up
+  the backup process.
+  (`#3608 <https://github.com/vertexproject/synapse/pull/3608>`_)
+- When a Cell is mirroring, gracefully go into read-only mode if the leader is
+  a greater version than the mirror.
+  (`#3581 <https://github.com/vertexproject/synapse/pull/3581>`_)
+- Add ``null`` as a constant that can be used in Storm expression syntax.
+  (`#3600 <https://github.com/vertexproject/synapse/pull/3600>`_)
+- Add ``cortex.storm.pool.get``, ``cortex.storm.pool.set`` and
```
Comment (Cisphyx) [inline CHANGELOG.rst:216]:
```suggestion
- Add ``cortex.storm.pool.get``, ``cortex.storm.pool.set``, and
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -6,6 +6,291 @@
...
+- Fix a bug where Cortex Cron jobs could start prior to data migrations
+  having completed running.
+  (`#3610 <https://github.com/vertexproject/synapse/pull/3610>`_)
+- Fix an error when granting a role admin permissions on a vault.
+  (`#3603 <https://github.com/vertexproject/synapse/pull/3603>`_)
+- Prevent the ``synapse.tools.easycert`` tool from making certificates with
+  names greater than 64 characters in length. Prevent AHA provisioning from
+  creating provisioning requests which would exceed that length.
+  (`#3609 <https://github.com/vertexproject/synapse/pull/3609>`_)
+- Fix an issue with the ``velocity`` base type not returning a float instead
```
Comment (Cisphyx) [inline CHANGELOG.rst:256]:
```suggestion
- Fix an issue with the ``velocity`` base type returning a float instead
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -6,6 +6,291 @@
...
+  of an integer when handling a string value without a unit.
+  (`#3616 <https://github.com/vertexproject/synapse/pull/3616>`_)
+- Fix an issue that could occur when pivoting from a secondary property to
+  a form when using variables for the source and target values.
+  (`#3618 <https://github.com/vertexproject/synapse/pull/3618>`_)
+- Fix a syntax parsing issue when using the try-set operator to set a node
+  property using a variable for the property name.
```
Comment (Cisphyx) [inline CHANGELOG.rst:263]:
```suggestion
- Fix a syntax parsing issue when using the try-set-plus or try-set-minus operator
  to update an array property on a node using a variable for the property name.
```

File: CHANGELOG.rst [hunk 7]
```diff
@@ -6,6 +6,291 @@
...
+  be removed in ``v3.0.0``.
+  (`#3589 <https://github.com/vertexproject/synapse/pull/3589>`_)
+- The Storm ``$lib.model.edge`` APIs have been marked as deprecated and will
+  be removed in ``v3.0.0``.
+  (`#3623 <https://github.com/vertexproject/synapse/pull/3623>`_)
+- The ``CoreAPI.enableMigrationMode()`` and ``CoreAPI.disableMigrationMode()``
+  Telepath methods have been marked as deprecated and will be removed after
+  2024-05-05.
+  (`#3610 <https://github.com/vertexproject/synapse/pull/3610>`_)
+- The Cortex configuration optoins ``cron:enable`` and ``trigger:enable`` have
```
Comment (Cisphyx) [inline CHANGELOG.rst:289]:
```suggestion
- The Cortex configuration options ``cron:enable`` and ``trigger:enable`` have
```

File: CHANGELOG.rst (hunk 2 above)
Comment (vEpiphyte) [reply CHANGELOG.rst:67]: Yup - already done.

**Review by Cisphyx — APPROVED** [review]

File: CHANGELOG.rst [hunk 8]
```diff
@@ -6,6 +6,308 @@
...
+- Fix a bug where Cortex Cron jobs could start prior to data migrations
+  having completed running.
+  (`#3610 <https://github.com/vertexproject/synapse/pull/3610>`_)
+- Fix an issue where ``node.prop.set`` and ``node.prop.del`` permissions were
+  not being properly checked.
+  (`#3627 <https://github.com/vertexproject/synapse/pull/3627>`_)
+- Fix a bug in the Storm ``merge`` command where the destination layer was
+  not being properly checked for property set and deletion permissions.
+  (`#3627 <https://github.com/vertexproject/synapse/pull/3627>`_)
+- Fix a bug in the Storm ``copyyto`` command where the destination layer was
```
Comment (Cisphyx) [inline CHANGELOG.rst:252]:
```suggestion
- Fix a bug in the Storm ``copyto`` command where the destination layer was
```

File: CHANGELOG.rst [hunk 9]
```diff
@@ -6,6 +6,308 @@
...
+  not being properly checked for property set operations.
```
Comment (Cisphyx) [inline CHANGELOG.rst:253]:
```suggestion
  not being properly checked for property set permissions.
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -6,6 +6,308 @@
...
+  of an integer when handling a string value without a unit.
+  (`#3616 <https://github.com/vertexproject/synapse/pull/3616>`_)
+- Fix an issue that could occur when pivoting from a secondary property to
+  a form when using variables for the source and target values.
+  (`#3618 <https://github.com/vertexproject/synapse/pull/3618>`_)
+- Fix a syntax parsing issue when using the try-set-plus or try-set-minus
+  operator to update an array property on a node using a variable for the
+  property name.
+  (`#3630 <https://github.com/vertexproject/synapse/pull/3630>`_)
+- Fix an issue with Aha service pools where their Telepath Clients were
```
Comment (Cisphyx) [inline CHANGELOG.rst:271]:
```suggestion
- Fix an issue with AHA service pools where their Telepath Clients were
```

File: CHANGELOG.rst [hunk 11]
```diff
@@ -6,6 +6,308 @@
...
+  not configured for use as ``aha://`` clients.
+  (`#3643 <https://github.com/vertexproject/synapse/pull/3643>`_)
+- Fix an isseue with Aha service pools where a fini'd Proxy was not properly
```
Comment (Cisphyx) [inline CHANGELOG.rst:274]:
```suggestion
- Fix an isseue with AHA service pools where a fini'd Proxy was not properly
```

File: CHANGELOG.rst [hunk 12]
```diff
@@ -6,6 +6,308 @@
...
+  Telepath methods have been marked as deprecated and will be removed after
+  2024-05-05.
+  (`#3610 <https://github.com/vertexproject/synapse/pull/3610>`_)
+- The Cortex configuration options ``cron:enable`` and ``trigger:enable`` have
+  been marked as deprecated and will be removed in ``v3.0.0``. These
+  configuration options no longer control cron or trigger behavior.
+  (`#3610 <https://github.com/vertexproject/synapse/pull/3610>`_)
+- The Storm Package  ``synapse_minversion`` key has been deprecated and will
+  be removed in ``v3.0.0``. Package authors should use the ``synapse_version``
+  key to specify a version range for packages they support. An example is
```
Comment (Cisphyx) [inline CHANGELOG.rst:307]:
```suggestion
  key to specify a version range for Synapse they support. An example is
```

File: CHANGELOG.rst [hunk 13]
```diff
@@ -6,6 +6,308 @@
...
+        The bank account which made the payment.
+
+      ``to:account``
+        The bank account which received the payment.
+
+      ``invoice``
+        The invoice that the payment applies to.
+
+      ``receipt``
+        The receipt that was issued for the payment.
```
Comment (Cisphyx) [inline CHANGELOG.rst:120]:
```suggestion
    ``from:account``
      The bank account which made the payment.

    ``to:account``
      The bank account which received the payment.

    ``invoice``
      The invoice that the payment applies to.

    ``receipt``
      The receipt that was issued for the payment.
```

File: CHANGELOG.rst [hunk 14]
```diff
@@ -6,6 +6,308 @@
...
+
+  ``file:mime:image``
+    The interface had the following property added to it:
+
+      ``text``
+        The text contained within the image.
```
Comment (Cisphyx) [inline CHANGELOG.rst:126]:
```suggestion
    ``text``
      The text contained within the image.
```

File: CHANGELOG.rst [hunk 15]
```diff
@@ -6,6 +6,308 @@
...
+
+  ``inet:email:message``
+    The form had the following property added to it:
+
+      ``flow``
+        The inet:flow which delivered the message.
```
Comment (Cisphyx) [inline CHANGELOG.rst:132]:
```suggestion
    ``flow``
      The inet:flow which delivered the message.
```

File: CHANGELOG.rst [hunk 16]
```diff
@@ -6,6 +6,308 @@
...
+    The form had the following property added to it:
+
+      ``mitre:attack:software``
+        A mapping to a MITRE ATT&CK software if applicable.
+
+  ``risk:mitigation``
+    The form had the following property added to it:
+
+      ``mitre:attack:mitigation``
+        A mapping to a MITRE ATT&CK mitigation if applicable.
```
Comment (Cisphyx) [inline CHANGELOG.rst:156]:
```suggestion
    ``issuer``
      The contact information of the office which issued the ID number.

  ``risk:threat``
    The form had the following property added to it:

    ``mitre:attack:group``
      A mapping to a MITRE ATT&CK group if applicable.

  ``risk:tool:software``
    The form had the following property added to it:

    ``mitre:attack:software``
      A mapping to a MITRE ATT&CK software if applicable.

  ``risk:mitigation``
    The form had the following property added to it:

    ``mitre:attack:mitigation``
      A mapping to a MITRE ATT&CK mitigation if applicable.
```

File: CHANGELOG.rst [hunk 17]
```diff
@@ -6,6 +6,308 @@
...
+  The following forms have been marked as deprecated:
+
+  ``risk:hasvuln``
+    Please use ``risk:vulnerable``.
+
+  **Light Edges**
+
+  ``has``
+
+    When used with a ``econ:bank:statement`` and a ``econ:acct:payment``, the
```
Comment (Cisphyx) [inline CHANGELOG.rst:169]:
```suggestion
    When used with an ``econ:bank:statement`` and an ``econ:acct:payment``, the
```

File: CHANGELOG.rst [hunk 18]
```diff
@@ -6,6 +6,308 @@
...
+    edge indicates the bank statement includes the payment.
+
+    When used with a ``ou:org`` node, the edge indicates the organization is
```
Comment (Cisphyx) [inline CHANGELOG.rst:172]:
```suggestion
    When used with an ``ou:org`` node, the edge indicates the organization is
```

File: CHANGELOG.rst [hunk 19]
```diff
@@ -6,6 +6,308 @@
...
+    was in possession of the target node.
+
+    When used with a ``sci:observation`` node, the edge indicates the
+    observations are summarized from the target nodes.
+
+    When used with an ``sci:evidence`` node, the edge indicates the evidence
+    includes observations from the target nodes.
+
+  ``owns``
+    When used with a ``ou:org`` node, the edge indicates the organization owns
```
Comment (Cisphyx) [inline CHANGELOG.rst:188]:
```suggestion
    When used with an ``ou:org`` node, the edge indicates the organization owns
```

File: CHANGELOG.rst [hunk 20]
```diff
@@ -6,6 +6,308 @@
...
+
+  The following forms have been marked as deprecated:
+
+  ``risk:hasvuln``
+    Please use ``risk:vulnerable``.
+
+  **Light Edges**
+
+  ``has``
+
```
Comment (Cisphyx) [inline CHANGELOG.rst:168]:
```suggestion
```

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**